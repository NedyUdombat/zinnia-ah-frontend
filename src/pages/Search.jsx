import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

//components
import Input from '../components/presentationals/Input/Input';
import Button from '../components/presentationals/Button/Button';
import Navbar from '../components/presentationals/Navbar/Navbar';

//stylesheets
import './Search.scss';

//modules
import { customSearch } from '../store/modules/search';
import Loader from '../components/presentationals/Loader/Loader';

export class Search extends Component {
  state = {
    keyword: '',
  };

  inputHandler = event => {
    const { value } = event.target;
    this.setState({ keyword: value });
  };

  submitHandler = event => {
    event.preventDefault();
    this.props.customSearch(this.state.keyword);
  };
  render() {
    let articles;

    if (this.props.articles.length > 0) {
      articles = this.props.articles.map(article => (
        <div key={article.id} className="article">
          Article
          <p key={article.title}>{article.title}</p>
          <p key={article.id}>{article.description}</p>
          <p key={article.id}>{article.body}</p>
          <p key={article.id}>{article.readTime}</p>
        </div>
      ));
    } else {
      articles = null;
    }
    let authors;

    if (this.props.authors.length > 0) {
      authors = this.props.authors.map(author => (
        <div key={author.username} className="author">
          Author
          <p key={author.firstName}>{author.firstName}</p>
          <p>{author.username}</p>
          <p>{author.lastName}</p>
        </div>
      ));
    } else {
      authors = null;
    }

    const loader = this.props.isLoading && (
      <Loader size="large" text="Loading... please wait" />
    );

    return (
      <Fragment>
        <Navbar />
        {loader}
        <div className="search-container">
          <form className="form" onSubmit={this.submitHandler}>
            <Input
              value={this.state.keyword}
              className="form-control"
              type="search"
              placeholder="Search Authors' Haven"
              onChange={this.inputHandler}
              name="q"
              data-test="search-input"
            />
            <Button
              className="btn-dark form-control"
              value="SEARCH"
              data-test="button"
            />
          </form>
        </div>
        <div className="articles-container">{articles}</div>
        <div className="authors-container">{authors}</div>
      </Fragment>
    );
  }
}

Search.propTypes = {
  articles: PropTypes.any,
  authors: PropTypes.any,
  customSearch: PropTypes.any,
  isLoading: PropTypes.bool,
  searchResult: PropTypes.array,
  users: PropTypes.any,
};

const mapStateToProps = state => ({
  articles: state.search.successResponse.data.articles,
  authors: state.search.successResponse.data.authors,
  isLoading: state.search.isLoading,
});

export default connect(
  mapStateToProps,
  { customSearch },
)(Search);
