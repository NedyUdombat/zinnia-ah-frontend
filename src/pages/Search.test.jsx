import React from 'react';
import { shallow } from 'enzyme';

import { Search } from './Search';

import { findByTestAttribute } from '../components/presentationals/SocialAuth/SocialAuth.test';

const component = shallow(<Search />);

describe('Custom Search', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders a search input field', () => {
    const wrapper = findByTestAttribute(component, 'search-input');
    expect(wrapper.length).toBe(1);
  });

  it('renders a search button', () => {
    const wrapper = findByTestAttribute(component, 'button');
    expect(wrapper.length).toBe(1);
  });
});
