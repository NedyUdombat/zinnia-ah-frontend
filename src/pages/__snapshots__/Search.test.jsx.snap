// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Search renders without errors 1`] = `
<Fragment>
  <Navbar
    profileUrl=""
  />
  <div
    className="search-container"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <Input
        className="form-control"
        data-test="search-input"
        name="q"
        onChange={[Function]}
        placeholder="Search Authors' Haven"
        type="search"
        value=""
      />
      <Button
        className="btn btn-dark form-control"
        data-test="button"
        value="SEARCH"
      />
    </form>
  </div>
  <Article />
</Fragment>
`;
